Sources/FrontEnd/TypeChecking/ConstraintGenerator.swift

```diff
@@ -14,8 +14,8 @@ struct ConstraintGenerator {
     /// The set of type constraints being generated.
     let constraints: [Constraint]
 
-    /// The diagnostics of the errors the generator encountered.
-    let diagnostics: [Diagnostic]
+    /// Indicates that the solver encountered one or more errors.
```

Comment:
And how does it indicate that? By being present?  By being `false`? ;-)

```suggestion
    /// True iff the solver encountered an error.
```

or if you must,

```suggestion
    /// True iff the solver encountered one or more errors.
```

or if you *truly* must

```suggestion
    /// Whether the solver encountered one or more errors.
```

Reply:
You don't tell me what "error" means in this context.  Without qualification, in programming, "error" usually means "postcondition failure" but I'm guessing this error is something more like a contradiction.  Why not use a more specific word?

Reply:
The property is no called `foundConflict` and the docs don't mention "errors" anymore.

I have to say, though, that it think it's reasonable to use the term "error" in the source code of a type checker to mean something else than a programming error of the type checker itself. I would not expect a component named "TypeChecker" or "ConstraintSolver" to set a flag when it encounters an error in its own implementation.